extensions/toggl-track/src/components/EditTimeEntryForm.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  useNavigation,
+  Form,
+  ActionPanel,
+  SubmitFormAction,
+  Icon,
+  showToast,
+  ToastStyle,
+  clearSearchBar,
+} from "@raycast/api";
+import toggl from "../toggl";
+import { storage } from "../storage";
+import { Project, TimeEntry } from "../toggl/types";
+import { useAppContext } from "../context";
+import { useMemo, useState } from "react";
+
+function EditTimeEntryForm({ project, entry }: { project?: Project; entry: TimeEntry }) {
+  const navigation = useNavigation();
+  const { projects, tags, isLoading, projectGroups } = useAppContext();
+  const [selectedProject, setSelectedProject] = useState<Project | undefined>(project);
+  const [selectedTags, setSelectedTags] = useState<string[]>([]);
+
+  async function handleSubmit(values: { description: string; tags: string[]; start: string; end?: string}) {
```

Comment:
No need to keep a selectedProject state, as we can get the project id from values

Reply:
Same goes for tags